src/ipa_pc/mod.rs

```diff
@@ -1346,23 +1363,30 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
             // z_i(x)
             let z_i_value = z_i_polynomial.evaluate(x_point);
 
-            // (p_i(X) - y_i)
-            let polynomial =
-                &(labeled_polynomial.polynomial() - &Polynomial::from_coefficients_slice(&[y_i]));
-
-            // LC(p_1(X),p_2(X),...,p_m(X)) = SUM ( lamda^i * z_i(x) * (p_i(X) - y_i) )
+            // LC(p_1(X),p_2(X),...,p_m(X)) = SUM ( lamda^i * z_i(x) * p_i(X) )
             lc_polynomial +=
-                (cur_challenge, &(polynomial * &Polynomial::from_coefficients_slice(&[z_i_value])));
+                (cur_challenge, &(labeled_polynomial.polynomial() * &Polynomial::from_coefficients_slice(&[z_i_value])));
```

Comment:
We also should do here a naive_mul, or use a scalar multiple function for dense polynomials, if it exists.

Reply:
Or, you do the multiplication of cur_challenge and z_i_value on the scalar part of the LC term. :)

Reply:
Done